docs/src/architecture/09_architecture_decisions/0004-app-loading.md

```diff
@@ -0,0 +1,40 @@
+---
+    title: 0004 Hermes application loading procedure
+    adr:
+        author: Alex Pozhylenkov <alex.pozhylenkov@iohk.io>
+        created: 11-Aug-2024
+        status:  draft
+---
+
+## Context
+
+Hermes as an engine which runs user's developed applications,
+obviously should have a functionality to load a and run provided assembled application.
+This procedure must validate application on the integrity and correctness,
+prepare application's state, before actually executing application itself.
```

Comment:
This would sound better
```suggestion
Hermes, as an engine that runs user-developed applications, should have the functionality to load and execute the provided assembled application. This process must validate the application's integrity and correctness and prepare the application's state before executing it.
```